### DISCUSSION TITLE
Custom root path?

### DISCUSSION THREAD

**Discussion by vago** [TanStack/router#1545] [Q&A] [answered]
Hi, I am fairly new to Tanstack router, but did read thru' the documentation. In our current setup, we have urls something like..

In development mode, 
`http://mysite.com/service/` - where `<service>` is dynamic value but fixed once we deploy the application

In production mode,
`http://service.mysite.com/`

Using file based routing, I added `routes/$service` directory to define route path, but it doesn't work in production setup since `service` is part of domain. 

Can you please suggest how I can handle this using file based routing? Thanks much

**Comment by SeanCassiere** [comment, marked as answer]
Are you referring to the [basepath](https://tanstack.com/router/latest/docs/framework/react/api/router/RouterOptionsType#basepath-property) configuration property?

You could set the basepath on the router when creating its instance using `createRouter`.

```tsx
const router = createRouter({
  routeTree,
  basePath: '/service'
})
```

**Reply by vago** [reply]
Indeed, this is what I was looking for. Should have dug more into the API, was just too excited. 
Thank you!
